app/controllers/payloads_controller.rb

```diff
@@ -1,9 +1,17 @@
 # Payload controller
 class PayloadsController < ApplicationController
   def index
-    @payloads = Payload.all
     if params[:nationality_id]
-      @payloads = @payloads.where(nationality_id: params[:nationality_id])
+      nationality = Nationality.find_by_id(params[:nationality_id])
+      if nationality
```

Comment:
This is an ok approach to handle 404. A better approach is to use a `before` filter in the controller and call a method that finds the resource. If the resource is not found, render 404 and the rest of the action gets never called. See https://guides.rubyonrails.org/action_controller_overview.html#filters

This is how it looks like:

```
class PayloadsController < ApplicationController

  before_action :load_nationality, except: [:show]
  before_action :load_payload, except: [:index]

  def index
    # your index action
  end

  def show
    # your show action
  end

  private

  def load_nationality
    @nationality = Nationality.find_by_id(params[:nationality_id])
    json: { success: false, message: 'Nationality not found' }, status: 404 and return unless @nationality
  end

  # same for payload

end
```